src/cgw_ucentral_messages_queue_manager.rs

```diff
@@ -0,0 +1,360 @@
+    // 3. Remove other message from qeueu and return it
+    pub async fn dequeue_device_message(
+        &self,
+        device_mac: &String,
+    ) -> Option<CGWUCentralMessagesQueueItem> {
+        let ret_msg: CGWUCentralMessagesQueueItem;
+        let default_msg = CGWUCentralCommand::default();
+        let container_lock = self.queue.read().await;
+
+        if self.get_device_messages_queue_len(device_mac).await == 0 {
+            return None;
+        }
+
+        let mut device_msg_queue = container_lock.get(device_mac).unwrap().write().await;
+        let reboot_msg = device_msg_queue
+            .get_item(MESSAGE_QUEUE_REBOOT_MSG_INDEX)
+            .unwrap()
+            .clone();
+        let configure_msg = device_msg_queue
+            .get_item(MESSAGE_QUEUE_CONFIGURE_MSG_INDEX)
+            .unwrap()
+            .clone();
+
+        if reboot_msg.parsed_message != default_msg {
+            ret_msg = reboot_msg;
+            device_msg_queue.remove_item(MESSAGE_QUEUE_REBOOT_MSG_INDEX);
+            device_msg_queue.insert_item(
+                MESSAGE_QUEUE_REBOOT_MSG_INDEX,
+                CGWUCentralMessagesQueueItem::default(),
+            );
+        } else if configure_msg.parsed_message != default_msg {
+            ret_msg = configure_msg;
+            device_msg_queue.remove_item(MESSAGE_QUEUE_CONFIGURE_MSG_INDEX);
+            device_msg_queue.insert_item(
+                MESSAGE_QUEUE_CONFIGURE_MSG_INDEX,
+                CGWUCentralMessagesQueueItem::default(),
+            );
+        } else {
+            ret_msg = device_msg_queue
+                .remove_item(MESSAGE_QUEUE_OTHER_MSG_INDEX)
+                .unwrap();
+        }
+
+        Some(ret_msg)
+    }
+
+    pub async fn get_device_queue_state(
+        &self,
+        device_mac: &String,
+    ) -> CGWUCentralMessagesQueueState {
+        let container_lock = self.queue.read().await;
+        let device_msg_queue = container_lock.get(device_mac).unwrap().read().await;
+
+        device_msg_queue.get_state()
+    }
+
+    pub async fn set_device_queue_state(
+        &self,
+        device_mac: &String,
+        state: CGWUCentralMessagesQueueState,
+    ) {
+        let container_lock = self.queue.read().await;
+        let mut device_msg_queue = container_lock.get(device_mac).unwrap().write().await;
+
+        device_msg_queue.set_state(state);
+    }
+
+    pub async fn insert_disconnected_device_timeout(&self, device_mac: String, seconds: u64) {
+        let mut disconnected_lock = self.disconnected_devices.write().await;
+        disconnected_lock.insert(device_mac, Duration::new(seconds, 0));
+    }
+
+    pub async fn remove_disconnected_device_timeout(&self, device_mac: &String) {
+        let mut disconnected_lock = self.disconnected_devices.write().await;
+        disconnected_lock.remove(device_mac);
+    }
+
+    pub async fn iterate_over_disconnected_devices(&self) {
```

Comment:
This function most likely doesn't need to be public?
Hence the naming for it can be the same (e.g. ignore comment above for line 302, do not rename it, but remove the pub)